sumpy/expansion/diff_op.py

```diff
@@ -145,7 +145,161 @@ def to_sym(self, fnames=None):
+    generates all the syzygies that when multiplied by the matrix gives a vector
+    with zeros except for the i-th component. Therefore the vector
+    $a_1, a_2, \ldots, a_n$ is in this module.
+
+    For each syzygy in the generating set, we calculate the inner product with the
+    first column to get a polynomial that represents a scalar PDE for the first
+    component. When there are multiple such scalar PDEs, we want to get a
+    combination of them that has the smallest positive degree.
+    To do this, we calculate a groebner basis of the polynomials. We use the
+    Groebner basis property that the largest monomial of each polynomial generated
+    by set of polynomials is divisible by the largest monomial of some polynomial
+    in the Groebner basis. When a graded monomial ordering is used this implies
+    that the degree of any polynomial generated is greater than or equal to the
+    degree of a polynomial in the Groebner basis. We choose that polynomial
+    as our scalar PDE.
+    """
+    import sympy
+    from sympy.polys.orderings import grevlex
+    gens = [sympy.symbols(f"_x{i}") for i in range(pde.dim)]
+    gens += [sympy.symbols(f"_t{i}") for i in range(pde.total_dims - pde.dim)]
+
+    max_vec_idx = 0
+    for eq in pde.eqs:
+        for deriv_ident in eq.keys():
+            max_vec_idx = max(max_vec_idx, deriv_ident.vec_idx)
+
+    pde_system_mat = sympy.zeros(len(pde.eqs), max_vec_idx + 1)
+    for row, eq in enumerate(pde.eqs):
+        for deriv_ident, coeff in eq.items():
+            deriv_as_poly = 1
+            for i, val in enumerate(deriv_ident.mi):
+                deriv_as_poly *= gens[i]**val
+            pde_system_mat[row, deriv_ident.vec_idx] += coeff * deriv_as_poly
+
+    ring = sympy.EX.old_poly_ring(*gens, order=grevlex)
+    column_ideals = [ring.free_module(1).submodule(*pde_system_mat[:, i].tolist(),
+                        order=grevlex)
+            for i in range(pde_system_mat.shape[1])]
+    column_syzygy_modules = [ideal.syzygy_module() for ideal in column_ideals]
+
+    ncols = len(column_syzygy_modules)
+
+    # For each column i, we need to get the intersection of all the syzygy modules
+    # except for the ith module. For n number of modules, this is $n*(n-2)$ work.
+    # To reduce that we first calculate the intersection from the left adding one
+    # module every iteration and store them. We then calculate the intersection
+    # from the right adding one module every iteration and store them. Then
+    # for each column we calculate the intersection of the left modules and the
+    # right modules. This requires only $3*(n-2)$ work.
+
+    # None represents the free module of the cartesian product of $m$ number of
+    # polynomials where $m$ is the number of rows of the PDE matrix.
+    left_intersections = [None]*ncols
+    right_intersections = [None]*ncols
+
+    def intersect(a, b):
+        if a is None:
```

Comment:
```suggestion
        # "None" serves as the neutral element.
        if a is None:
```
(or get rid of the `None` handling?)

Reply:
Done in 54943a3ed74f7669563c8645576810161b145968